src/util/base-input.ts

```diff
@@ -255,9 +262,20 @@ export class BaseInput<T> extends Ion implements CommonInput<T> {
    */
   hasValue(): boolean {
     const val = this._value;
-    return isArray(val)
-      ? val.length > 0
-      : isPresent(val);
+    if (!isPresent(val)) {
+      return false;
+    }
+    if (isArray(val) || isString(val)) {
+      return val.length > 0;
+    }
+    return true;
+  }
+
+  /**
+   * @hidden
+   */
+  focusNext() {
```

Comment:
where is focusNext() used? need to review it
how does it interact between input, toggle, checkbox (look at items/inputs)